### ISSUE TITLE
[a11y]: Dropdown and multi-select dropdown inconsistent and have implementation issues

### ISSUE THREAD

**Issue by mbgower** [carbon-design-system/carbon#12721]
### Package

@carbon/react

### Browser

Chrome

### Operating System

MacOS

### Package version

React library

### React version

https://react.carbondesignsystem.com/?path=/story/components-dropdown--default

### Automated testing tool and ruleset

n/a

### Assistive technology

n/a

### Description

There are interaction differences and challenges between the React Dropdown and [Multi-select dropdown](https://react.carbondesignsystem.com/?path=/story/components-multiselect--playground) 

### Issue 0 Inconsistent categorization between Storybook and documentation
I originally just mentioned this in passing, but I think it is worth flagging as an issue: Multi-select is a separate component in Storybook (under M), but is just a variation on Dropdown in the Carbon guidance and on the Figma design tool. I think it should consistently be co-located in Storybook as well. If there is concern with the amount of samples under one topic, you could always call it Dropdown Multiselect so that it as least appears next in the left navigation.

### Issue 1 Inconsistent keystrokes
For both, they should open on `Enter`, `Spacebar`, `Down Arrow` and (optionally) `Alt+Down Arrow` (`Cmd+Down Arrow` on Mac). 
This is the behaviour for the Dropdown (good), but the Multi-select only opens on Enter and Spacebar (bad). Especially because unselected dropdowns/multiselects look the same, they should operate the same.

### Issue 2 Inconsistent behaviour on open
When they open, the currently shown option should be given focus /highlighted in the dropdown list (see the first screenshot under issue 3 for what this should look like) .
This is not the behaviour for any of them.
For the Dropdown, if the **Enter or Spacebar** is selected, the default value "Dropdown menu options" is not even displayed in the list, so nothing in the list takes focus. (Multi-select has the same behaviour.)
![image](https://user-images.githubusercontent.com/14298245/203630075-7e7159cc-67a4-4151-ad7e-d453be3d1455.png)
#### Issue 2b
For the Dropdown, if **Down Arrow or Alt+Down Arrow** is issued, the _next_ item in the list is selected on open. This is also incorrect. This is likely a result of Carbon's implementation of a pseudo option/placeholder (see issue 3 below). The ARIA Authoring Practices, although not containing a dropdown equivalent, do suggest that for a combobox, the Alt+Down arrow combination should open the list _without_ moving into the options. That is potentially; however, I think I would also consider it fine if either with or without the modifier, it opened and highlighted the first option.
![image](https://user-images.githubusercontent.com/14298245/203630365-35160eba-aa3b-4479-add5-7778c54200b4.png)

### Issue 3 There is no placeholder in a dropdown; Every shown value should be an option. 
As shown in the [anatomy](https://carbondesignsystem.com/components/dropdown/usage/#anatomy) guidance, the label precedes the input, and the only thing that can appear in the input field is an option.
![image](https://user-images.githubusercontent.com/14298245/203640553-0157ac1a-01be-4482-b3ac-0bb08e3825c2.png)

The Dropdown largely gets this correct. However, the starting value is not an option, and so when the component is opened nothing gets focus (see first screen shot in Issue 2). Same thing happens in Multi-select.

Note also that the required label value in storybook does not appear to hold the "Dropdown label" value shown in the implementation, but appears blank in the Controls section.
![image](https://user-images.githubusercontent.com/14298245/203640307-1912647c-d118-43b8-b426-0a11c9aabc6f.png)

---

Note that there _is_ confusion on the idea of placeholder text in the guidance itself (pictured below). I think it would be better to NEVER use placeholder here, but rather 'default option' to avoid confusion with placeholder in a text input. But I will open an issue against the guidance website for that. In the meantime, if both of these React storybook implementations can properly use default option values, it will clear up a lot of confusion.
![image](https://user-images.githubusercontent.com/14298245/203643529-bdf4b1a7-e950-4bcd-a253-ed622f008671.png)

### Issue 4 The implementation confuses/mislabel the title, label,
The multi-select markedly departs from the terminology of Dropdown. First it swaps out the "label" text preceding the input and calls it titleText. You can see the "This is MultiSelect Title" specified with the value in this screen shot. The instructions _do_ say "Provide text to be used in a `label` element..." but it is singularly unhelpful to change the terminology.

![image](https://user-images.githubusercontent.com/14298245/203642640-e3c3fe0b-e521-4f6d-b4d6-546b90c6c5a9.png)

The added confusion is made clear when we see what they have called the default option value, "label", here given a value of "This is a label". Note that the control description here -- "Generic `label`..." -- is _extremely_ confusing.

![image](https://user-images.githubusercontent.com/14298245/203640017-b1e2c186-40f7-4dc9-931e-882c412df46a.png)

This is also the case in the Live demo area; an attribute called "label" which should be the default option. As noted elsewhere, this default option _can_ have the effect of being a placeholder, but it also remains the top option in the list. 

<img width="599" alt="image" src="https://user-images.githubusercontent.com/14298245/211894416-c9bc9f6e-8134-4db5-bcdc-c15db78399eb.png">

**These variables should be renamed to match those in the Dropdown anatomy section.**

Finally, I'll mention that I have a separate issue  https://github.com/carbon-design-system/carbon/issues/12723 tracking _just_ multiselect challenges (not shared by the basic dropdown)

### Reproduction/example

https://react.carbondesignsystem.com/?path=/story/components-multiselect--playground

### Steps to reproduce

I will be updating the issue with more content...

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

```[tasklist]
### Tasks
- [ ] Inconsistent categorization between Storybook and documentation Ref #12947
- [ ] https://github.com/carbon-design-system/carbon/pull/13097
- [ ] https://github.com/carbon-design-system/carbon/issues/13096
- [ ] https://github.com/carbon-design-system/carbon/issues/13185
```

**Comment by alisonjoseph** [comment]
> Issue 0 Inconsistent categorization between Storybook and documentation

> I originally just mentioned this in passing, but I think it is worth flagging as an issue: Multi-select is a separate component in Storybook (under M), but is just a variation on Dropdown in the Carbon guidance and on the Figma design tool. I think it should consistently be co-located in Storybook as well. If there is concern with the amount of samples under one topic, you could always call it Dropdown Multiselect so that it as least appears next in the left navigation.

This is being addressed as part of a much larger issue where we are looking across all components across website, storybook and figma, but agreed they should be (and will be) grouped together. @mbgower

**Comment by tw15egan** [comment]
>Issue 4 The implementation confuses/mislabel the title, label,

I agree; this trips me up as well. It needs to be renamed as part of the v12 effort and should be aligned with all other form/input elements. Unfortunately, we cannot change the name now as it would be considered breaking.

**Comment by tay1orjones** [comment]
Related #13185

**Comment by mbgower** [comment]
@sstrubberg another one where I would suggest that as much as I don't like the differences in behaviour here, I do not think this rates the same priority as a few other obvious failures/bugs.
